src/back-end/lib/resources/session.ts

```diff
@@ -68,36 +47,42 @@ export async function signOut(
   return valid(session);
 }
 
-const resource: Resource = {
-  routeNamespace: "sessions",
+const routeNamespace = "sessions";
+
+const readOne: crud.ReadOne<Session, Connection> = (
+  _connection: Connection
+) => {
+  return nullRequestBodyHandler(async (request) => {
+    const respond = (code: number, body: Session | string[]) =>
+      basicResponse(code, request.session, makeJsonResponseBody(body));
+    if (!permissions.readOneSession(request.session, request.params.id)) {
+      return respond(401, [permissions.ERROR_MESSAGE]);
+    }
+    return respond(200, request.session);
+  });
+};
 
-  readOne(connection) {
-    return nullRequestBodyHandler(async (request) => {
-      const respond = (code: number, body: Session | string[]) =>
-        basicResponse(code, request.session, makeJsonResponseBody(body));
-      if (!permissions.readOneSession(request.session, request.params.id)) {
-        return respond(401, [permissions.ERROR_MESSAGE]);
-      }
-      return respond(200, request.session);
-    });
-  },
+const delete_: crud.Delete<Session, Connection> = (connection: Connection) => {
```

Comment:
I used underscore as a suffix here as a workaround to TypeScript not allowing me to name it `delete`. The intention is for it to be read as "delete prime". Other languages allow `delete'` as the more common convention, but since TypeScript doesn't allow that here, I've opted for `delete_`. Note that `delete_` is not exported out of this module, so it's use would be limited in the code base. I'm happy to adopt a different convention if one is preferred though.